client/state/marketplace/selectors.ts

```diff
@@ -67,4 +70,16 @@ export const hasOrIntendsToBuyLiveSupport = ( state: IAppState ): boolean => {
 	return hasLiveSupport;
 };
 
-export default shouldUpgradeCheck;
+export function canPublishProductReviews(
+	state: IAppState,
+	productType: string,
+	productSlug: string
+) {
+	if ( productType === 'theme' ) {
+		return canPublishThemeReview( state, productSlug );
+	}
+	if ( productType === 'plugin' ) {
+		throw new Error( `Selector nnt implemented for plugins` );
+	}
```

Comment:
For plugins we can use a hook called `useCanPublishPluginReview`. In the ideal scenario both would be a hook, but as Themes use class components this mismatch happens.

That said I wouldn't bother with `canPublishProductReviews` and directly use each method when necessary.

Reply:
The problem is that the Reviews component is agnostic to the product type so a generic selector would encapsulate the logic instead of using different hooks that also cannot be conditionals (because of the rules of hooks). 
A selector works fine in this scenario and it can be enhanced if other products are added without needing to modify the Ratings component

Reply:
> The problem is that the Reviews component is agnostic to the product type 

Aha, I didn't know this part, this is very nice. 

In this case, you are right; it might be a good idea to extract that logic from the hook and add it to a selector.